### PR TITLE
feat(function): switch to function, add renovate, bump providers

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- enable renovate
- bump configurations
- bump up

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
--- PASS: kuttl (1005.69s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (1005.20s)
PASS
17:18:41 [ OK ] running automated tests
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,9 +61,20 @@ build.init: $(UP)
...
 	@$(INFO) running automated tests
-	@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) $(UPTEST) e2e examples/postgres-aws-claim.yaml,examples/mariadb-aws-claim.yaml,examples/postgres-azure-claim.yaml,examples/mariadb-azure-claim.yaml --setup-script=test/setup.sh --default-timeout=2400 || $(FAIL)
+	@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) CROSSPLANE_NAMESPACE=$(CROSSPLANE_NAMESPACE) $(UPTEST) e2e examples/postgres-aws-claim.yaml,examples/mariadb-aws-claim.yaml,examples/postgres-azure-claim.yaml,examples/mariadb-azure-claim.yaml --setup-script=test/setup.sh --default-timeout=2400 || $(FAIL)
 	@$(OK) running automated tests
 
 e2e: build controlplane.up local.xpkg.deploy.configuration.$(PROJECT_NAME) uptest
 
-.PHONY: uptest e2e
+render:
+	crossplane beta render examples/mariadb-aws-claim.yaml apis/aws/composition.yaml examples/function/function.yaml -r
```
Comment (ytsarev) [inline Makefile:70]: nit: should we maybe just drop `examples/function/function.yaml` to `examples/functions.yaml` for simplicity?

Comment (haarchri) [reply Makefile:70]: think its safer if someone is doing a `kubectl apply -f examples/ `- it will not install the function again - ?!

Comment (ytsarev) [reply Makefile:70]: it should be idempotent

**Review by ytsarev — APPROVED** [review]
I will not hold this pr, function.yaml location is minor

**Pull request merged**